src/NServiceBus.NHibernate/TimeoutPersisters/IncorrectIndexDetector.cs

```diff
@@ -0,0 +1,120 @@
+﻿namespace NServiceBus.TimeoutPersisters.NHibernate.Installer
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Data.Common;
+    using System.Linq;
+    using global::NHibernate.Cfg;
+    using global::NHibernate.Dialect;
+    using global::NHibernate.Mapping;
+    using global::NHibernate.Tool.hbm2ddl;
+    using NServiceBus.Logging;
+
+    class IncorrectIndexDetector
+    {
+        static readonly ILog Logger = LogManager.GetLogger(typeof(IncorrectIndexDetector));
+
+        public IncorrectIndexDetector(Configuration configuration)
+        {
+            this.dialect = Dialect.GetDialect(configuration.Properties);
+            this.connectionHelper = new ManagedProviderConnectionHelper(MergeProperties(configuration.Properties));
+            this.timeoutEntityMapping = configuration.GetClassMapping(typeof(TimeoutEntity));
+        }
+
+        Dictionary<string, string> MergeProperties(IDictionary<string, string> properties)
+        {
+            var props = new Dictionary<string, string>(dialect.DefaultProperties);
+            foreach (var prop in properties)
+            {
+                props[prop.Key] = prop.Value;
+            }
+            return props;
+        }
+
+        public void LogWarningIfTimeoutEntityIndexIsIncorrect()
+        {
+            try
+            {
+                connectionHelper.Prepare();
+                var connection = connectionHelper.Connection;
+
+                var index = GetIndex(connection, timeoutEntityMapping, "TimeoutEntity_EndpointIdx");
+                if (index == null)
+                {
+                    Logger.Warn("Could not find TimeoutEntity_EndpointIdx index definition. Consult NServiceBus NHibernate persistence documentation for details on how to create this index.");
```

Comment:
Those definitions make sense once the endpoint has started and is actually processing messages but this is part of startup I'm fine with WARN as the level but are we never logging stuff as ERROR during startup?